solr/core/src/java/org/apache/solr/search/facet/HLLAgg.java

```diff
@@ -79,18 +79,18 @@ public SlotAcc createSlotAcc(FacetContext fcontext, long numDocs, int numSlots)
         return new SortedNumericAcc(fcontext, getArg(), numSlots);
       } else if (sf.hasDocValues()) {
         return new UniqueMultiDvSlotAcc(
-            fcontext, sf, numSlots, fcontext.isShard() ? factory : null);
+            fcontext, sf, numSlots, 0, fcontext.isShard() ? factory : null);
       } else {
         return new UniqueMultivaluedSlotAcc(
-            fcontext, sf, numSlots, fcontext.isShard() ? factory : null);
+            fcontext, sf, numSlots, 0, fcontext.isShard() ? factory : null);
       }
     } else {
       if (sf.getType().getNumberType() != null) {
         // always use hll here since we don't know how many values there are?
         return new NumericAcc(fcontext, getArg(), numSlots);
       } else {
         return new UniqueSinglevaluedSlotAcc(
-            fcontext, sf, numSlots, fcontext.isShard() ? factory : null);
+            fcontext, sf, numSlots, 0, fcontext.isShard() ? factory : null);
```

Comment:
why do these default to `0`? Is it b/c of the way HLL works, passing explicit values for deduping is not applicable?

Reply:
I feel like I had good reasons originally but looking at this now and I don't think this is necessarily right. Still looking though to double check.

Reply:
Ah I see why this is safe (although open to adjustments if you have better ideas, could just pass the default even if not used). When `factory` is passed to any of these unique implementations (which it is for all shard handling), it triggers the shard value that is produced to take a special path where it produces hll bytes based on the hll impl produced by the factory. The code paths to return explicit values is never used and really doesn't make sense for hll. But I do admit, there is a bit of a smell here relying on this.

Reply:
👍 thanks for checking, let me know what you come up with. I can't really see a reason not to upstream this. It's straightforward, and even fulfills an existing `TODO` in the upstream code. Ideally looking for ways to reduce the overhead of managing a custom fork.